### PR TITLE
Feature/328 logout for login dot gov

### PR DESCRIPTION
**Pull request by dheitzer**

### CODE REVIEW

**Comment by dheitzer** [comment]
This one is ready for CR

**Review by Elaine-Krauss-TCG — APPROVED** [review]

**Comment by MitchellTCG** [comment]
@toddlees Can you please CR this pull?

File: front-end/src/app/shared/services/api.service.ts [hunk 1]
```diff
@@ -48,6 +48,14 @@ export class ApiService {
     return this.http.get<T>(`${environment.apiUrl}${endpoint}`, { headers: headers, params: params, withCredentials: true });
   }
 
+  public getAbsoluteUrl<T>(
```
Comment (toddlees) [inline front-end/src/app/shared/services/api.service.ts]: I think the purpose of the api url is that it's for interactions with the (FECfile) api.  Requests to other api's should be in their own services

File: front-end/src/app/shared/services/login.service.ts [hunk 2]
```diff
@@ -66,12 +66,10 @@ export class LoginService {
...
       this.store.dispatch(userLoggedOutAction());
     } else {
-      this.apiService.postAbsoluteUrl(`${environment.loginDotGovLogoutUrl}`, null).pipe(
-        tap(() => {
-          this.clearUserLoggedInCookies();
-          this.store.dispatch(userLoggedOutAction());
-        })
-      ).subscribe(() => undefined);
+      this.cookieService.delete('csrftoken');
+      window.location.href = environment.loginDotGovLogoutUrl || "";
```
Comment (toddlees) [inline front-end/src/app/shared/services/login.service.ts]: Seems like this is the fix 👍 .  As for the code before, `pipe(tap(...))` [is for side effects](https://rxjs.dev/api/operators/tap) and is redundant to the subscribe below where line 71 and 72 would be more appropriate

**Review by toddlees — APPROVED** [review]

File: front-end/src/app/shared/services/login.service.ts (hunk 2 above)
Comment (toddlees) [inline front-end/src/app/shared/services/login.service.ts]: This seems to be causing the unit tests to fail.  Maybe there's a way to turn this off when it's running unit tests

**Review by toddlees — CHANGES_REQUESTED** [review]

**Review by toddlees — APPROVED** [review]

**Pull request merged**